src/Piral.Blazor.Core/Program.cs

```diff
@@ -24,7 +24,8 @@ public static async Task Main(string[] args)
             .AddSingleton(new HttpClient { BaseAddress = baseAddress })
             .AddSingleton<IComponentActivator, PiletComponentActivator>()
             .AddSingleton<IComponentActivationService, ComponentActivationService>()
-            .AddSingleton<IModuleContainerService, ModuleContainerService>();
+            .AddSingleton<IModuleContainerService, ModuleContainerService>()
+            .AddSingleton<IQueryParameterValueProvider, QueryParameterValueProvider>();

```

Comment:
It's a viable approach, but why don't we just use the existing approach and extend it? Right now we already wrap / infer the parameters. We can just extend that functionality to look for the `PiralQueryParameterAttribute` and apply the correct value when needed.

Maybe I am missing something here. Thanks already!

Reply:
I would like to extend the existing code. Unfortunately, I have no idea what is passed in the IDictionary<string, JsonElement>. 

How can I debug the core lib code?

I would like to implement it cleanly in the existing way.

Reply:
That sounds very good.
I'm looking forward to the next release. 
Thank you very much   😀